pkg/api/grafana_com_proxy.go

```diff
@@ -24,8 +29,101 @@ var grafanaComProxyTransport = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
-func ReverseProxyGnetReq(logger log.Logger, proxyPath string, version string) *httputil.ReverseProxy {
-	url, _ := url.Parse(setting.GrafanaComUrl)
+func (hs *HTTPServer) ListGnetPlugins(c *models.ReqContext) {
+	proxy := hs.ReverseProxyGnetReq(c.Logger, "/plugins")
+	proxy.Transport = grafanaComProxyTransport
+	director := proxy.Director
+	proxy.Director = func(r *http.Request) {
+		// Call director set with ReverseProxyGnetReq
+		director(r)
+		// Request no encoding to simplify modifying the response
+		r.Header.Set("Accept-Encoding", "")
+	}
+	writeResponse := func(r *http.Response, b []byte) {
+		r.Body = io.NopCloser(bytes.NewReader(b))
+		r.ContentLength = int64(len(b))
+		r.Header.Set("Content-Length", strconv.Itoa(len(b)))
+	}
+	proxy.ModifyResponse = func(r *http.Response) error {
+		if hs.AccessControl.IsDisabled() {
+			return nil
+		}
+
+		defer func() { _ = r.Body.Close() }()
+		b, errRead := io.ReadAll(r.Body)
```

Comment:
It would be interesting to refactor this part of the code to do the check per plugin instead of waiting for the whole body to be received. Something like this: https://stackoverflow.com/a/42612995.

Nothing we need to do in this PR but could be worth considering for the future.